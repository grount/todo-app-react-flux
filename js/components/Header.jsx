import React from 'react';
import '../../css/index.css';

export default class Header extends React.Component {
	constructor(props) {
		super(props);
		this.handleMouseDown = this.handleMouseDown.bind(this);
		this.handleMouseUp = this.handleMouseUp.bind(this);
	}

	handleMouseDown(e) {
		e.target.className = 'nav-button active';
	}

	handleMouseUp(e) {
		e.target.className = 'nav-button';
	}

	render() {
		return (
			<nav className="nav-bar">
				<p id="logo">Todo-Application</p>
				<button 
					type="button"
					className="nav-button"
					id="clear-button"
					onClick={this.props.handleClearAllOption}
					onMouseDown={this.handleMouseDown}
					onMouseUp={this.handleMouseUp}>
					Clear All
				</button>

				<button 
					type="button"
					className="nav-button" 
					id="clear-selected-button" 
					onClick={this.props.handleClearSelectedOptions}
					onMouseDown={this.handleMouseDown}
					onMouseUp={this.handleMouseUp}>
					Clear Selected 
				</button>
			</nav>
		);
	}
}
