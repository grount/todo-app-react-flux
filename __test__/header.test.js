
import React from 'react';
import { shallow, mount, render } from 'enzyme';
import toJson from 'enzyme-to-json'
import Header from '../js/components/Header.jsx';


test('Should render properly', () => {
	const component = shallow(<Header />);
	const tree = toJson(component);
	expect(tree).toMatchSnapshot();
});
