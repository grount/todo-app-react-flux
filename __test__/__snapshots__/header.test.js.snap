// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should render properly 1`] = `
<nav
  className="nav-bar"
>
  <p
    id="logo"
  >
    Todo-Application
  </p>
  <button
    className="nav-button"
    id="clear-button"
    onMouseDown={[Function]}
    onMouseUp={[Function]}
    type="button"
  >
    Clear All
  </button>
  <button
    className="nav-button"
    id="clear-selected-button"
    onMouseDown={[Function]}
    onMouseUp={[Function]}
    type="button"
  >
    Clear Selected
  </button>
</nav>
`;
